vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -78,69 +132,176 @@
                     /* get current date with old parsing */
                     currentDate = datepicker.i18n.parseDate(inputValue);
                 }
 
                 /* create test-string where to extract parsing regex */
-                let testDate = new Date(Date.UTC(datepicker.$connector.yearPart.initial, datepicker.$connector.monthPart.initial - 1, datepicker.$connector.dayPart.initial));
+                let testDate = new Date(Date.UTC(parseInt(yearPart.initial), parseInt(monthPart.initial) - 1, parseInt(dayPart.initial)));
                 let testString = cleanString(testDate.toLocaleDateString(locale, { timeZone: 'UTC' }));
-                datepicker.$connector.parts.forEach(function (part) {
+                parts.forEach(function (part) {
                     part.index = testString.indexOf(part.initial);
                 });
                 /* sort items to match correct places in regex groups */
-                datepicker.$connector.parts.sort(FlowDatePickerPart.compare);
+                parts.sort(FlowDatePickerPart.compare);
                 /* create regex
-                * regex will be the date, so that:
-                * - day-part is '(\d{1,2})' (1 or 2 digits),
-                * - month-part is '(\d{1,2})' (1 or 2 digits),
-                * - year-part is '(\d{4})' (4 digits)
-                *
-                * and everything else is left as is.
-                * For example, us date "10/20/2010" => "(\d{1,2})/(\d{1,2})/(\d{4})".
-                *
-                * The sorting part solves that which part is which (for example,
-                * here the first part is month, second day and third year)
-                *  */
-                datepicker.$connector.regex = testString.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&').replace(datepicker.$connector.dayPart.initial, "(\\d{1,2})").replace(datepicker.$connector.monthPart.initial, "(\\d{1,2})").replace(datepicker.$connector.yearPart.initial, "(\\d{4})");
-
-                datepicker.i18n.formatDate = tryCatchWrapper(function (date) {
+                 * regex will be the date, so that:
+                 * - day-part is '(\d{1,2})' (1 or 2 digits),
+                 * - month-part is '(\d{1,2})' (1 or 2 digits),
+                 * - year-part is '(\d{1,4})' (1 to 4 digits)
+                 *
+                 * and everything else is left as is.
+                 * For example, us date "10/20/2010" => "(\d{1,2})/(\d{1,2})/(\d{1,4})".
+                 *
+                 * The sorting part solves that which part is which (for example,
+                 * here the first part is month, second day and third year)
+                 */
+                datepicker.$connector.regex = testString.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&')
+                    .replace(dayPart.initial, "(\\d{1,2})")
+                    .replace(monthPart.initial, "(\\d{1,2})")
+                    .replace(yearPart.initial, "(\\d{1,4})");
```

Comment:
yearPart regex was changed from `\\d{4}` to `\\d{1,4}` to allow input of 1-4 digit years instead of only exactly 4 digit years (it's still converted to 4 digits when formatting).